Documentation/artifact-format.md

```diff
@@ -175,8 +259,85 @@ version is specified outside of `header.tar.gz`.
 
 Why is there a tar file inside a tar file? See the "Ordering" section.
 
+### header-info (version 3)
+
+Format: JSON
+
+`header-info` must be the first file within `header.tar.gz`. Its content is:
+
+```
+{
+  "updates": [
+    {
+      "type": "rootfs-image"
+    },
+    {
+      "type": "rootfs-image"
+    },
+    {
+      ...
+    }
+  ],
+  "artifact_provides": [
+          { "artifact_name": "name" },
+          { "artifact_group": "group-1" },
+          { "update_types_supported": ["rootfs-image"]},
+          { "provides_type": ["provides_value", "provides_value"] },
+      ],
```

Comment:
I think in this list we need to have `rootfs_image_checksum`, do you agree? This will be the final checksum, whereas the one in the manifest will be either for the delta, or for the original full rootfs `0000.tar.gz` (which is not identical to the final sum, due to tar and compression).

This also means that I think the full rootfs artifact needs to be "delta enabled" by having the payload outside the manifest, otherwise the delta update cannot remove the original full rootfs without breaking the checksum.